### DISCUSSION TITLE
Page Number subtracted by 1

### DISCUSSION THREAD

**Discussion by mcrexy** [eclipse-birt/birt#1799] [Q&A] [answered]
Hi,

My coverpage is supposed to start at page 1. However, we won't show this numbering and wants the next page to start at 1 onwards.

Is there a way to do this? Autotext is always counting from the first page. 

I have tried a variable but it's not working well, especially with multiple tables having page break in betweens.

**Comment by hvbtup** [comment]
I don't think this is possible with BIRT, but maybe someone else knows a solution.

If your report is generated in PDF format and you do not present it to the user directly (that is, if you have integrated BIRT into your application), there are workaround possible.
For example, you could generate your cover page with a second report (*). Your main report then only generated the content pages (without the cover page). Then you have got two PDFs. You can then use OpenPDF (part of the BIRT libraries) to "merge" the PDFs into one PDF.

(*) This second report could also be the same rptdesign file, with a report parameter that decides if only the cover page or only the content pages should be generated.

**Reply by mcrexy** [reply]
Thank you @hvbtup for the sharing. 

Whoa, is there any example I can look at on this OpenPDF on same rptdesign file?

But not sure if my "total page" on coverpage would be affected.

**Comment by hvbtup** [comment]
To make this clear: You need to add some Java code to your integration. An example for this (but without BIRT) can be found at https://github.com/LibrePDF/OpenPDF/blob/master/pdf-toolbox/src/test/java/com/lowagie/examples/general/copystamp/Concatenate.java

**Reply by speckyspooky** [reply]
For the concatination the PDF-emitter of BIRT himselsf supports the merge to.
For that the user-property "PdfEmitter.AppendDocumentList" kann be used.

You can take a look here:
https://github.com/eclipse-birt/birt/blob/ed3df844276f95ac59b1b82276e7b25cbaab0e2f/engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/README.md

**Reply by mcrexy** [reply]
Hello @hvbtup, thank you for sharing more on the Java!

**Comment by speckyspooky** [comment, marked as answer]
There is a way how you can handle it within report with multiple tables an so on.
For that I created a standard report and the master page got an dynamic-text.
The special thing is the JavaScript on the onRender()-method of the element.
There is a small script added to count according your role without the cover.
![grafik](https://github.com/user-attachments/assets/00638e18-e71a-4182-bdbd-bfa1a7deaa2b)


I added my test report which is not very beautiful but I think it can help you.
Please be aware you need the Run and render task to get the result. If you start the PDF-creation directly from BIRT-eclipse then the pages won't be calculated. If the PDF-created from viewer so the document is created in full with page numbering the PDF looks fine.

### PDF-result
[cover-content-pages.pdf](https://github.com/user-attachments/files/16273305/cover-content-pages.pdf)

### Demo-report

[cover-content-pages.zip](https://github.com/user-attachments/files/16273314/cover-content-pages.zip)

**Reply by mcrexy** [reply]
@speckyspooky Wow, thanks a lot! I just had the time to revisit this issue and realised I had done something similar to you! Just that I always render straight from BIRT-Eclipse, didn't know about this till you mentioned it.

Thank you guys for the assistance and elaborated examples!

**Comment by mcrexy** [comment]
Hi @speckyspooky,

By any chance, have you used ReportServer.net before?

I am facing same rendering issues. I have emailed them on this but it seems they can't have a reply on the matter.

**Comment by speckyspooky** [comment]
No, I have never used "ReportServer.net". The included BIRT version would be the same like from our open source project.
It could be that the BIRT version is a little bit older like the latest version from here.

I use as commercial security-framework servBIRD (from [Tradui](https://tradui.de) ) but there will be used the same BIRT version from here.

**Reply by mcrexy** [reply]
I see, thank you for the sharing on servBIRD!
